kitsune/search/tests/test_api.py

```diff
@@ -65,6 +67,30 @@ def test_invalid_product(self):
         eq_(res.status_code, 400)
         eq_(res.data['detail'], {'product': 'Could not find product with slug "nonexistant".'})
 
+    def test_invalid_locale(self):
+        res = self.client.get(reverse('search.suggest'), {
+            'locale': 'bad-medicine',
+            'q': 'search',
+        })
+        eq_(res.status_code, 400)
+        eq_(res.data['detail'], {'locale': 'Could not find locale "bad-medicine".'})
+
+    def test_invalid_fallback_locale(self):
+        for locale, fallback in settings.NON_SUPPORTED_LOCALES.items():
+            if fallback is not None:
+                break
```

Comment:
My bad--I'm remembering wrong. It's the for in list comprehensions that leaks in Python 2, but not Python 3.

https://docs.python.org/3/whatsnew/3.0.html#changed-syntax